src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -1236,29 +1272,26 @@ impl PredicatePushdown for LogicalAgg {
 
 impl ToBatch for LogicalAgg {
     fn to_batch(&self) -> Result<PlanRef> {
-        let new_input = self.input().to_batch()?;
+        self.to_batch_with_order_required(&Order::any())
+    }
+
+    fn to_batch_with_order_required(&self, required_order: &Order) -> Result<PlanRef> {
+        let mut input_order = Order::any();
+        let output_requires_order = self.output_requires_order_on_group_keys(required_order);
```

Comment:
We can do better here. if the output is ordered by a, it is also ordered by (a,b) and any order key whose prefix is a(you can find it in `Order::satisfies`). For example, a SQL `SELECT max(v1), k1, k2 from t group by k1, k2 order by k1`. The required Order here is Order(k1), we can require the Agg's input is ordered by (k1, k2). then the sortAgg's output is also ordered by(k1, k2) which also satisfies ordered by(k1)